testproject/Assets/Tests/Runtime/MultiprocessRuntime/BaseMultiprocessTests.cs

```diff
@@ -101,14 +101,19 @@ public virtual IEnumerator Setup()
             // Moved this out of OnSceneLoaded as OnSceneLoaded is a callback from the SceneManager and just wanted to avoid creating
             // processes from within the same callstack/context as the SceneManager.  This will instantiate up to the WorkerCount and
             // then any subsequent calls to Setup if there are already workers it will skip this step
-            if (MultiprocessOrchestration.Processes.Count < WorkerCount)
+            if (MultiprocessOrchestration.ActiveWorkerCount() < WorkerCount)
             {
-                var numProcessesToCreate = WorkerCount - MultiprocessOrchestration.Processes.Count;
+                var numProcessesToCreate = WorkerCount - MultiprocessOrchestration.ActiveWorkerCount();
                 for (int i = 0; i < numProcessesToCreate; i++)
                 {
+                    MultiProcessLog($"Spawning testplayer {i} since {MultiprocessOrchestration.ActiveWorkerCount()} is less than {WorkerCount}");
                     MultiprocessOrchestration.StartWorkerNode(); // will automatically start built player as clients
                 }
             }
+            else
+            {
+                MultiProcessLog($"No need to spawn a new test player as there are already existing processes {MultiprocessOrchestration.ActiveWorkerCount()}");
```

Comment:
thinking of it, how come we could already have workers running? Does this mean we aren't launching new players for each test fixture?
There's a danger with this, if tests don't properly cleanup their state, this could influence following runs. Especially for perf tests. Clearing players had the advantage of starting from a clean state for each fixture.
I understand we have resource contention. I'm guessing that change comes for this?

Reply:
I was trying to keep the logic as close to the original as possible. This was always how it was, clients were not disposed between test points and were reused across test points. I (along with @NoelStephensUnity ) tried to change the logic but ran into stability issues as the launch of a new client actually takes quite a while to process.

I'd actually prefer a clean, fresh client for each test point but the scope of that change is turning out to be quite large. We can tackle it here but I'd prefer to tackle that in a separate PR.

Reply:
oh this is in setup. I think I misread and thought it was in SetupSuite.
Then, it's business as usual, ok, forget my comment